thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java

```diff
@@ -309,7 +309,9 @@ public Long runBackfill(long functionId, DateTime backfillStartTime, DateTime ba
     }
 
     BackfillKey backfillKey = new BackfillKey(functionId, backfillStartTime, backfillEndTime);
-    Thread returnedThread = existingBackfillJobs.putIfAbsent(backfillKey, Thread.currentThread());
```

Comment:
The logic of putIfAbsent works differently to the logic you are replacing with. It should be:

Thread returnedThread = null;
if (existingBackfillJobs.containsKey(backfillKey) {
  returnedThread = existingBackfillJobs.get(backfillKey);
} else {
  existingBackfillJobs.put(backfillKey, Thread.currentThread());
}